x-pack/test/security_functional/plugins/test_endpoints/server/init_routes.ts

```diff
@@ -39,6 +39,32 @@ export function initRoutes(
   );
 
   const router = core.http.createRouter();
+
+  for (const isAuthFlow of [true, false]) {
+    router.get(
+      {
+        path: `/authentication/app/${isAuthFlow ? 'auth_flow' : 'not_auth_flow'}`,
+        validate: {
+          query: schema.object({
+            statusCode: schema.maybe(schema.number()),
+            message: schema.maybe(schema.string()),
+          }),
+        },
+        options: { tags: isAuthFlow ? ['security:authFlow'] : [] },
```

Comment:
__nit:__ just to keep it closer to real auth-flow routes that don't require authentication
```suggestion
        options: { tags: isAuthFlow ? ['security:authFlow'] : [], authRequired: !isAuthFlow },
```

Also, would be great to add a code comment for these routes to clarify why we added them and how they can be useful. There is no doubt, we'll soon forget why we even added them :slightly_smiling_face: